src/views/checkups/CheckupsRunButton.tsx

```diff
@@ -1,72 +1,81 @@
-import { trimLastHistoryPath } from './utils/utils';
+import { CHECKUP_URLS } from './utils/constants';
+import { getCurrentCheckupType, trimLastHistoryPath } from './utils/utils';
 
 const CheckupsRunButton: FC = () => {
   const [namespace] = useActiveNamespace();
   const navigate = useNavigate();
   const location = useLocation();
   const { t } = useKubevirtTranslation();
+
   const { isPermitted: isCreateNetworkPermitted } = useCheckupsNetworkPermissions();
   const { isPermitted: isCreateStoragePermitted } = useCheckupsStoragePermissions();
 
-  const createItems = {
-    network: (
-      <div
-        className={classNames({ 'CheckupsRunButton--item__disabled': !isCreateNetworkPermitted })}
-      >
-        {t('Network latency')}
-      </div>
-    ),
-    storage: (
-      <div
-        className={classNames({ 'CheckupsRunButton--item__disabled': !isCreateStoragePermitted })}
-      >
-        {t('Storage')}
-      </div>
-    ),
+  const currentCheckupType = useMemo(
+    () => getCurrentCheckupType(location.pathname),
+    [location.pathname],
+  );
+
+  const isDisabled = useMemo(() => {
+    if (ALL_NAMESPACES_SESSION_KEY === namespace) {
+      return true;
+    }
+
+    switch (currentCheckupType) {
+      case CHECKUP_URLS.NETWORK:
+        return !isCreateNetworkPermitted;
+      case CHECKUP_URLS.STORAGE:
+        return !isCreateStoragePermitted;
+      case CHECKUP_URLS.SELF_VALIDATION:
+        // Self-validation is handled by SelfValidationCheckupRunButton
+        return true;
+      default:
+        return true;
+    }
+  }, [namespace, currentCheckupType, isCreateNetworkPermitted, isCreateStoragePermitted]);
+
+  const handleRunCheckup = () => {
+    if (isDisabled || !currentCheckupType) return;
+
+    const basePath = trimLastHistoryPath(location.pathname);
+    switch (currentCheckupType) {
+      case CHECKUP_URLS.NETWORK:
+        navigate(createURL(`${CHECKUP_URLS.NETWORK}/form`, basePath));
+        break;
+      case CHECKUP_URLS.STORAGE:
+        navigate(createURL(`${CHECKUP_URLS.STORAGE}/form`, basePath));
+        break;
+      // Self-validation is handled by SelfValidationCheckupRunButton
+    }
   };
 
-  const onCreate = useCallback(
-    (type: string) => {
-      switch (type) {
-        case 'network':
-          return (
-            isCreateNetworkPermitted &&
-            navigate(createURL('network/form', trimLastHistoryPath(location.pathname)))
-          );
-        case 'storage':
-          return (
-            isCreateStoragePermitted &&
-            navigate(createURL('storage/form', trimLastHistoryPath(location.pathname)))
-          );
-      }
-    },
-    [isCreateNetworkPermitted, navigate, location, isCreateStoragePermitted],
-  );
+  if (currentCheckupType === CHECKUP_URLS.SELF_VALIDATION) {
+    return <SelfValidationCheckupRunButton />;
+  }
 
   return (
     <Button
       className={classNames({
         'CheckupsRunButton--main': ALL_NAMESPACES_SESSION_KEY === namespace,
```

Comment:
do we still need that extra styling? we already set `isDisabled` flag for all-namespaces.